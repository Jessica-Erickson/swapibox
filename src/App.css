.App {
  background-color: black;
  display: flex;
  flex-flow: row wrap;
  text-align: center;
}
