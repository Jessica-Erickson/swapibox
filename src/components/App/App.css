.App {
  height: 100vh;
  width: 100vw;
  display: grid;
  grid-template-columns: 33% 1fr;
  grid-template-rows: 35% 1fr;
  grid-row-gap: 40px;
  font-family: 'News Cycle', sans-serif;
  font-weight: 400;
}

.loading {
  height: 100vh;
  width: 100vw;
  background-image: url(../../assets/images/bb8-loading.gif);
  background-position: center;
  background-repeat: no-repeat;
  background-color: rgb(8, 31, 41)
}

h1,
h2,
h3,
h4 {
  font-weight: 700;
}