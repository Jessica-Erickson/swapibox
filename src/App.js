import React, { Component } from 'react';
import './App.css';
import Card from './Card';

class App extends Component {
  constructor() {
    super();
    this.state = {
      cards: [1,2,3,4,5,6,7,8,9,10]
    }
  }

  render() {
    return (
      <div className="App">
      {
        this.state.cards.map((element, index) => {
          return <Card key={index}/>
        })
      }
      </div>
    );
  }
}

export default App;
