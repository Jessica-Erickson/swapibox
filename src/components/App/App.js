import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import * as API from '../../helper.js';
import ScrollingText from '../ScrollingText';
import Header from '../Header';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      favorites: [],
      filmsLoading: true,
      peopleLoading: true,
      planetsLoading: true,
      vehiclesLoading: true,
    }
  }

  componentDidMount = () => {
    if (!localStorage.getItem('favorites')) {
      const favorites = JSON.parse(localStorage.getItem('favorites'));
      this.setState({ favorites });
    }
  }

  loadingCheck = (category) => {
    this.setState({ [category]: false });
  }

  render() {
    const { filmsLoading, peopleLoading, planetsLoading, vehiclesLoading, favorites } = this.state;
    const ready = !filmsLoading && !peopleLoading && !planetsLoading && !vehiclesLoading;

    return (
      <div className={ ready ? 'App' : 'loading' }>
        <ScrollingText
          display={ready}
          loadingCheck={() => {
            loadingCheck('filmsLoading');
          }} />
        <Header 
          display={ready}
          favorites={favorites.length} />
        <Route exact path='/' render={() => {
          <h2 className="default">Select a Category or Favorites</h2>
        }}>
        <Route exact path='/favorites' render={() => {
          <CardContainer 
            favorites={favorites}
            cardContents='favorites' />
        }} />
        <Route exact path='/people' render={() => {
          <CardContainer 
            favorites={favorites}
            cardContents='people' />
        }} />
        <Route exact path='/planets' render={() => {
          <CardContainer 
            favorites={favorites}
            cardContents='planets' />
        }} />
        <Route exact path='/vehicles' render={() => {
          <CardContainer 
            favorites={favorites}
            cardContents='vehicles' />
        }} />
      </div>
    )
  }
}

export default App;