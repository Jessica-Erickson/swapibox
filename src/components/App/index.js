import React, { Component } from 'react';
import * as API from '../../helper.js';

import ScrollingText from '../ScrollingText'
import Header from '../Header'
import CardContainer from '../CardContainer'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      isLoading: true,
      default: [],
      allFilms: [],
      people: [],
      favorites: [],
      currentDisplay: 'default'
    }
  }

  async componentDidMount() {
    this.setState({ 
      allFilms: await API.getFilms(),
      people: await API.getPeople(), 
      isLoading: false });
  }

  render() {
    const { isLoading, allFilms, favorites, currentDisplay } = this.state;
    const cardContents = this.state[currentDisplay];

    if (isLoading) {
      return (
        <div className="loading">
        </div>
      )
    } else {
      return (
        <div className="App">
          <ScrollingText allFilms={allFilms} />
          <Header favorites={favorites} />
          <CardContainer cardContents={cardContents} />
        </div>
      );
    }
  }
}

export default App;
