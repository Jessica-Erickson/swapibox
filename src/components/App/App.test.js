import React from 'react';
import { shallow } from 'enzyme';
import App from './index';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true })

    expect(wrapper.state()).toMatchSnapshot()
  })
})