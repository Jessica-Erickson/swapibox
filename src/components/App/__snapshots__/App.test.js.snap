// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="loading"
/>
`;

exports[`App should add favorites to state and localStorage when favorited 1`] = `
Object {
  "allFilms": Array [],
  "currentDisplay": "default",
  "default": Array [],
  "favorites": Array [
    Object {
      "cardData": "someData",
    },
  ],
  "isLoading": true,
  "people": Array [],
  "planets": Array [],
  "vehicles": Array [],
}
`;

exports[`App should getStorage on refresh / subsequent mounts 1`] = `
Object {
  "allFilms": Array [
    Object {},
  ],
  "currentDisplay": "default",
  "default": Array [],
  "favorites": Array [
    Object {},
  ],
  "isLoading": false,
  "people": Array [
    Object {},
  ],
  "planets": Array [
    Object {},
  ],
  "vehicles": Array [
    Object {},
  ],
}
`;

exports[`App should have a default state 1`] = `
Object {
  "allFilms": Array [],
  "currentDisplay": "default",
  "default": Array [],
  "favorites": Array [],
  "isLoading": true,
  "people": Array [],
  "planets": Array [],
  "vehicles": Array [],
}
`;

exports[`App should remove favorites from state and localStorage when unfavorited 1`] = `
Object {
  "allFilms": Array [],
  "currentDisplay": "default",
  "default": Array [],
  "favorites": Array [],
  "isLoading": true,
  "people": Array [],
  "planets": Array [],
  "vehicles": Array [],
}
`;

exports[`App should render the app if not Loading 1`] = `
<div
  className="App"
>
  <ScrollingText
    allFilms={Array []}
  />
  <Header
    currentDisplay="default"
    favorites={0}
    handleNavClick={[Function]}
  />
  <CardContainer
    addFavorite={[Function]}
    cardContents={Array []}
    currentDisplay="default"
    removeFavorite={[Function]}
  />
</div>
`;

exports[`App should set favorites to [] if no favorites in localStorage 1`] = `
Object {
  "allFilms": Array [
    Object {},
  ],
  "currentDisplay": "default",
  "default": Array [],
  "favorites": Array [],
  "isLoading": false,
  "people": Array [
    Object {},
  ],
  "planets": Array [
    Object {},
  ],
  "vehicles": Array [
    Object {},
  ],
}
`;
